File: DSA/data_structures/linked_list/linkedListTraversal.js
class Node {
	constructor(val) {
		this.val = val
		this.next = null
	}
}

const a = new Node('A')
const b = new Node('B')
const c = new Node('C')
const d = new Node('D')

a.next = b
b.next = c
c.next = d

const printLinkedList = (head) => {
	let current = head
	while (current != null) {
		console.log(current.val)
		current = current.next
	}
}

const printLinkedListRec = (head) => {
	if (head === null) return
	console.log(head.val)
	printLinkedListRec(head.next)
}

console.log('Printing linked list')
printLinkedList(a)
console.log('Printing linked list recursively')
printLinkedListRec(a)

File: dsa/linked_list/linked_list_traversal.py
class Node:
    def __init__(self, val):
        self.val = val
        self.next = None

a = Node('A')
b = Node('B')
c = Node('C')
d = Node('D')

a.next = b
b.next = c
c.next = d

def print_linked_list(head):
    current = head
    while current:
        print(current.val)
        current = current.next

def print_linked_list_rec(head):
    if head == None:
        return 
    print(head.val, end='->')
    print_linked_list_rec(head.next)

if __name__ == '__main__':
    print('Printing linked list')
    print_linked_list(a)
    print('Printing linked list recursively')
    print_linked_list_rec(a)
